dot/sync/disjoint_block_set_test.go

```diff
@@ -15,6 +16,9 @@ import (
 
 func TestDisjointBlockSet(t *testing.T) {
 	s := newDisjointBlockSet(pendingBlocksLimit)
+	s.timeNow = func() time.Time {
+		return time.Time{}
+	}
```

Comment:
nit you could use a constant reference point, i.e.

```go
	const nowUnix = 1000
	now := time.Unix(nowUnix, 0)
	s.timeNow = func() time.Time {
		return now
	}

	// Further down you can use `now` instead of `time.Time{}`
```

That helped me out when writing tests for heavily time oriented things. Not really the case here but a good way to write time tests I think :wink:

Reply:
I think  `time.Time{}` is essentially time 0 - the problem I found with using `time.Unix` was that the other cached values inside the `time.Time` struct didn't match when doing the `Equal` check, causing tests to fail

Reply:
Yes it's time 0.

That's a strange problem, I never seen this. There is the location information in time.Time but it shouldn't matter. Let me know if you encounter this problem again, I'd be curious!